rpi-update

```diff
@@ -735,23 +752,37 @@ if [[ ! -f "${FW_REVFILE}" ]]; then
 else
 	if [[ "${ARTIFACT}" != "" ]]; then
 		LOCAL_HASH=$(cat "${FW_REVFILE}")
+		BOOTLOADER_LOCAL_HASH=$(cat "${BOOTLOADER_REVFILE}")
 	else
-		LOCAL_HASH=$(get_long_hash "$(cat "${FW_REVFILE}")")
+		LOCAL_HASH=$(get_long_hash ${REPO_API_URI} "$(cat "${FW_REVFILE}")")
+		BOOTLOADER_LOCAL_HASH=$(get_long_hash "${BOOTLOADER_REPO_API_URI}"  "$(cat "${BOOTLOADER_REVFILE}")")
 	fi
-	if [[ "${LOCAL_HASH}" == "${FW_REV}" ]]; then
-		echo " *** Your firmware is already up to date (delete ${FW_REVFILE} to force an update anyway)"
+	if [[ "${LOCAL_HASH}" == "${FW_REV}" && "${BOOTLOADER_LOCAL_HASH}" == "${BOOTLOADER_REV}" ]] ; then
```

Comment:
If either firmware or bootloader needs to update then the update should run. I've modified the check above so that if SKIP_BOOTLOADER is set then BOTOLOADER_LOCAL_HASH will be the same as BOOTLOADER_REV

I guess a bootloader-only update might cause some unnecessary updating of firmware + kernel?

Reply:
Yes - that was my thought. Maybe bootloader updates will be rare enough that this isn't a big problem, but it might be worth skipping a firmware+kernel update if the hash says no change is needed.

Reply:
Ok, will take a look at that. Need to check whether it's download_rev or elsewhere.

I also changed it to not treat the hash being identical to be classified as a 'downgrade'

Reply:
Added a fixup commit that attempts to skip firmware downloading + updates of related components if only the bootloader has changed. Might be cleaner if there was a higher level flag instead of multiple SKIPS but I didn't see anything obvious